CreateService/handler.ts

```diff
@@ -0,0 +1,216 @@
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import {
+  ObjectIdGenerator,
+  ulidGenerator
+} from "io-functions-commons/dist/src/utils/strings";
+import { EmailString, NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Service } from "../generated/api-admin/Service";
+import { Subscription } from "../generated/api-admin/Subscription";
+import { ServicePayload } from "../generated/definitions/ServicePayload";
+import { ServiceWithSubscriptionKeys } from "../generated/definitions/ServiceWithSubscriptionKeys";
+import { APIClient } from "../utils/clients/admin";
+import { getLogger, ILogger } from "../utils/logging";
+import {
+  ErrorResponses,
+  IResponseErrorUnauthorized,
+  toDefaultResponseErrorInternal,
+  toErrorServerResponse
+} from "../utils/responses";
+
+type ResponseTypes =
+  | IResponseSuccessJson<ServiceWithSubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+const logPrefix = "CreateServiceHandler";
+
+/**
+ * Type of a CreateService handler.
+ *
+ * CreateService expects a service payload as input
+ * and returns service with subscription keys
+ */
+type ICreateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const createSubscriptionTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  userEmail: EmailString,
+  subscriptionId: NonEmptyString,
+  productName: NonEmptyString
+): TaskEither<ErrorResponses, Subscription> =>
+  tryCatch(
+    () =>
+      apiClient.createSubscription({
+        email: userEmail,
+        productNamePayload: {
+          product_name: productName
+        },
+        subscription_id: subscriptionId
+      }),
+    errs => {
+      logger.logUnknown(errs);
+      return toDefaultResponseErrorInternal(errs);
+    }
+  ).foldTaskEither(
+    err => fromLeft(err),
+    errorOrResponse =>
+      errorOrResponse.fold(
+        errs => {
+          logger.logErrors(errs);
+          return fromLeft(toDefaultResponseErrorInternal(errs));
+        },
+        responseType =>
+          responseType.status !== 200
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(responseType.value)
+      )
+  );
+const createServiceTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  subscriptionId: NonEmptyString
+): TaskEither<ErrorResponses, Service> =>
+  tryCatch(
+    () =>
+      apiClient.createService({
+        service: {
+          ...servicePayload,
+          authorized_recipients: [],
```

Comment:
```suggestion
          authorized_recipients: [sandboxFiscalCode],
```

You need to take sandboxFiscalCode from an ENV var named SANDBOX_FISCAL_CODE.

Reply:
I think that we should not provide a default value for this field (create / update) and just let the caller specify a value like with the other fields

Reply:
@gunzip @francescopersico: `ServicePayload` does not contain informations about `authorized_recipients`. Do we have to modify `ServicePayload` definition?

Reply:
Fixed in [2f63ab6](https://github.com/pagopa/io-functions-services/pull/68/commits/2f63ab68bf93f2db5b11b603e52fa36b04de3306)